examples/md-flexible/src/ParticleSerializationTools.cpp

```diff
@@ -5,41 +5,81 @@
  */
 #include "ParticleSerializationTools.h"
 
-namespace ParticleSerializationTools {
-void serializeParticle(const ParticleType &particle, std::vector<char> &serializedParticles) {
-  ParticleAttributes attributes{.position{particle.getR()},
-                                .velocity{particle.getV()},
-                                .force{particle.getF()},
-                                .id = particle.getID(),
-                                .ownershipState = particle.getOwnershipState(),
-                                .typeId = particle.getTypeId(),
-                                .oldForce{particle.getOldF()}};
+#include <tuple>
+
+namespace {
+
+constexpr std::array<typename ParticleType::AttributeNames, 15> Attributes = {
+    ParticleType::AttributeNames::id,
+    ParticleType::AttributeNames::posX,
+    ParticleType::AttributeNames::posY,
+    ParticleType::AttributeNames::posZ,
+    ParticleType::AttributeNames::velocityX,
+    ParticleType::AttributeNames::velocityY,
+    ParticleType::AttributeNames::velocityZ,
+    ParticleType::AttributeNames::forceX,
+    ParticleType::AttributeNames::forceY,
+    ParticleType::AttributeNames::forceZ,
+    ParticleType::AttributeNames::oldForceX,
+    ParticleType::AttributeNames::oldForceY,
+    ParticleType::AttributeNames::oldForceZ,
+    ParticleType::AttributeNames::typeId,
+    ParticleType::AttributeNames::ownershipState};
+
+constexpr size_t AttributesSize = 120;
```

Comment:
Yes, the automation is currently commented out. I don't think I can use the `SoAArraysType` because it includes the pointer to the particle. Using `Particle::get()` would work ( see commented code ). Sadly I cannot make the commented sizeOfAttributes static to avoid recalculating the size. So I decided it is to much overhead to recalculate the size for every particle every time.